libs/ui-watt/src/lib/components/tooltip/watt-tooltip.directive.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-disable @angular-eslint/no-host-metadata-property */
+import { AriaDescriber, FocusMonitor } from '@angular/cdk/a11y';
+import { Directionality } from '@angular/cdk/bidi';
+import {
+  Overlay,
+  ScrollDispatcher,
+  ScrollStrategy,
+} from '@angular/cdk/overlay';
+import { Platform } from '@angular/cdk/platform';
+import { DOCUMENT } from '@angular/common';
+import {
+  Directive,
+  ElementRef,
+  Inject,
+  Input,
+  NgZone,
+  Optional,
+  ViewContainerRef,
+} from '@angular/core';
+import {
+  MatTooltip,
+  MatTooltipDefaultOptions,
+  MAT_TOOLTIP_DEFAULT_OPTIONS,
+  MAT_TOOLTIP_SCROLL_STRATEGY,
+  TooltipPosition,
+} from '@angular/material/tooltip';
+
+export type wattTooltipPosition = TooltipPosition;
+
+@Directive({
+  selector: '[wattTooltip]',
+  exportAs: 'wattTooltip',
+  host: {
+    class: 'mat-mdc-tooltip-trigger',
```

Comment:
question: is this in order to be forwards compatible 😁

Reply:
Haha, I guess 😄 I basically just copied the entire decorator from the internal implementation. Do you think I should remove it?

Reply:
If it should be here, then it must probably be just `mat-tooltip-trigger` because that's used in the [current version](https://github.com/angular/components/blob/main/src/material/legacy-tooltip/tooltip.ts#L48) (renamed to "legacy" in the upcoming v15) of the tooltip component. The `mat-mdc-tootip-trigger` class is used in the "new" version (based on MDC). 

But I think we can remove it and the right class will still be applied by the parent class (hope so).